colossus/src/main/scala/colossus/protocols/http/StreamService.scala

```diff
@@ -0,0 +1,222 @@
+package colossus
+package protocols.http
+package stream
+
+import controller._
+import service.Protocol
+import core._
+import service._
+
+import scala.language.higherKinds
+import scala.util.{Try, Success, Failure}
+
+trait StreamingHttpMessage[T <: HttpMessageHead] {
+
+  def head: T
+  def body: Source[Data]
+
+  def collapse: Source[HttpStream[T]] = Source.one[HttpStream[T]](Head(head)) ++ body
+}
+
+case class StreamingHttpRequest(head: HttpRequestHead, body: Source[Data]) extends StreamingHttpMessage[HttpRequestHead]
+case class StreamingHttpResponse(head: HttpResponseHead, body: Source[Data]) extends StreamingHttpMessage[HttpResponseHead]
+
+
+trait StreamingHttp extends Protocol {
+  type Request = StreamingHttpMessage[HttpRequestHead]
+  type Response = StreamingHttpMessage[HttpResponseHead]
+}
+
+
+object GenEncoding {
```

Comment:
Supposed to mean "generic".  I will probably find a better name and also try to figure out if this needs to be public